test/python/dialects/indexing/dialect.py

```diff
@@ -439,3 +440,37 @@ def testSimpleLiteralIndexing():
     print(Tensor(w.owner.operands[1]))
     # CHECK: %{{.*}} = indexing.gather %[[TEN]][%[[CST0]]] gather_dims([0, 1, 2]) unique : (tensor<10x10x10x10xi32>, tensor<3xindex>) -> tensor<10xi32>
     print(w.owner)
+
+
+# CHECK-LABEL: TEST: testCanonicalizeTupleIndexCastListLiteral
+@run
+def testCanonicalizeTupleIndexCastListLiteral():
+  with mlir_mod_ctx() as module:
+
+    for n_tens in range(1, 4):
+      uniqs = set()
+      n_slices = 4 - n_tens
+      ten_idx = [[0], [1]]
+      slice_idx = slice(None)
+      for p in permutations([str(ten_idx)] * n_tens +
```

Comment:
A more pythonic way would be `uniqs = set(permuations(...))`. (Though you probably don't even need the `set`, do you?)

Reply:
You do need set because you want to shuffle `ten_idx` around but you want the `n_tens` copies to distinguishable.

Reply:
Ah, right, some of the permutations will equivalent because some elements of the original list are equivalent...